llvm/lib/CodeGen/MachineScheduler.cpp

```diff
@@ -780,7 +780,7 @@ void MachineSchedulerBase::scheduleRegions(ScheduleDAGInstrs &Scheduler,
       Scheduler.enterRegion(&*MBB, I, RegionEnd, NumRegionInstrs);
 
       // Skip empty scheduling regions (0 or 1 schedulable instructions).
```

Comment:
Adjust the comment to explain why this shouldn't skip 1 sized regions. Also adjust the patch description